eva-accession-release-automation/run_release_in_embassy/create_release_tracking_table.py

```diff
@@ -0,0 +1,305 @@
+            'num_rs_to_release int8 null, '       # not computed but still used by release automation
+            'total_num_variants int8 null, '      # not computed and unused
+            'release_folder_name text null, '
+            'release_status text null, '
+            'primary key (taxonomy, assembly_accession, release_version))'
+        )
+        execute_query(self.metadata_conn, query_create_table)
+
+    def fill_release_entries(self):
+        """Fill in release table based on previous release data, EVA metadata, and supported assembly tracker.
+        Also fills in should_be_released values."""
+        self._fill_from_previous_release()
+        self._fill_from_eva_metadata()
+        self._fill_from_supported_assembly_tracker()
+        self.fill_should_be_released_for_all()
+
+    def _fill_from_previous_release(self):
+        query = f"""select taxonomy, scientific_name, assembly_accession, sources, fasta_path, report_path, 
+                    release_folder_name from eva_progress_tracker.clustering_release_tracker 
+                    where release_version = {self.release_version - 1}"""
+        for tax, sc_name, asm_acc, src, fs_path, rpt_path, rls_folder_name in get_all_results_for_query(
+                self.metadata_conn, query):
+            self._insert_entry_for_taxonomy_and_assembly(tax, asm_acc, src, sc_name, fs_path, rpt_path,
+                                                         rls_folder_name)
+
+    def _fill_from_eva_metadata(self):
+        query = f"""select distinct  pt.taxonomy_id as taxonomy, asm.assembly_accession as assembly_accession
+                    from evapro.project_taxonomy pt
+                    join evapro.project_analysis pa on pt.project_accession = pa.project_accession 
+                    join evapro.analysis a on a.analysis_accession = pa.analysis_accession 
+                    join evapro.assembly asm on asm.assembly_set_id = a.assembly_set_id 
+                    and asm.assembly_accession is not null and assembly_accession like 'GCA%'"""
+        sources = 'EVA'
+        for tax, asm_acc in get_all_results_for_query(self.metadata_conn, query):
+            self._insert_entry_for_taxonomy_and_assembly(tax, asm_acc, sources)
+
+    def _fill_from_supported_assembly_tracker(self):
+        query = f"""select distinct taxonomy_id as taxonomy, assembly_id as assembly_accession
+                    from evapro.supported_assembly_tracker"""
+        sources = 'DBSNP, EVA'
+        for tax, asm_acc in get_all_results_for_query(self.metadata_conn, query):
+            self._insert_entry_for_taxonomy_and_assembly(tax, asm_acc, sources)
+
+    def _insert_entry_for_taxonomy_and_assembly(self, tax, asm_acc, sources, sc_name=None, fasta_path=None,
+                                                report_path=None, release_folder_name=None):
+        sc_name = sc_name if sc_name else get_scientific_name_from_ensembl(tax)
+        sc_name = sc_name.replace("'", "\''")
```

Comment:
change to accommodate our good friend `Ambystoma 'unisexual hybrid'`

Reply:
Ticket to remove scientific name where it's not needed: [EVA-3405](https://www.ebi.ac.uk/panda/jira/browse/EVA-3405)